### PR TITLE
Add linear moment scaling to coupe and meca

### PR DESCRIPTION
**Pull request by PaulWessel**
This PR implements Torsten Becker's [suggestion](https://forum.generic-mapping-tools.org/t/linear-scaling-of-moment-tensor-symbols-for-psmeca/1978/4) of an option to scale symbol sizes proportional to moment and not just magnitude.  We do this by introducing some common constants in _meca.h_ that sets the default earthquake parameters and then use those in **coupe** and **meca**.  Changing mode and setting parameters are done via new modifiers to **-S**.  We also deprecated **-M**. Summary of changes:

1. Change from magnitude to moment scaling via **-S** modifier **+l**.
2. Set reference magnitude or moment via **-S** modifier **+s**_reference_.
3. Set fixed symbol size via **-S** modifier **+m** (deprecated option **-M** handled via backward compatibility).
4. Update test scripts to use the new syntax after first passing with the old syntax.  All tests pass.
5. Documentation has been updated to reflect the changes.

Currently, the default reference parameters are these (in meca.h):

```
#define SEIS_MAG_REFERENCE 5		/* Reference magnitude for -S */
#define SEIS_MOMENT_MANT_REFERENCE 4	/* Mantissa for reference moment for -S */
#define SEIS_MOMENT_EXP_REFERENCE 23	/* Exponent for reference moment for -S */
```

@seisman may revise these and possibly add corresponding SI units versions for the moment attributes so that we can upgrade the reference value to take a unit, e.g. **-S...+s**1e26**u** (for US unit or whatever).

### CODE REVIEW

**Comment by PaulWessel** [comment]
Successful test by Becker on the [forum](https://forum.generic-mapping-tools.org/t/linear-scaling-of-moment-tensor-symbols-for-psmeca/1978/14).  Once I can add his test example this PR will be complete.

**Comment by PaulWessel** [comment]
I have finalized this PR by addressing Becker's comments about the integer constants needing to be float, and added a new test script _seis_13.sh_ which produces the plot below.  I think we are ready for final approval:

seis_13

File: test/seis/seis_13.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+34.48	29.31	15	-0.242	-2.082	2.324	0.727	-0.911	-2.959	24	34.48	29.31	B112395C
+34.58	28.45	15	0.034	-0.508	0.474	0.737	0.107	-0.586	24	34.58	28.45	C201605160145A
+34.62	28.83	28.3	-0.102	-1.69	1.8	0.3	-0.246	-2.04	24	34.62	28.83	C201506271534A
+34.64	27.13	12	-0.207	-1.1	1.3	0.106	-0.004	-0.334	24	34.64	27.13	C202006161430A
+37.76	25.2	12	-3.99	1.59	2.4	0.354	-0.285	-1.85	24	37.76	25.2	C200905191735A
+34.4	28.62	15	-1.325	-0.131	1.457	-0.173	-0.831	-0.683	25	34.4	28.62	C080393G
+34.73	29.07	18.4	-1.361	-3.255	4.616	3.887	0.091	-4.493	26	34.73	29.07	M112295A
+EOF
+gmt begin seis_13
+	sort -gr +9 tmp.txt | gmt psmeca -Sm0.4i+f0+l+s5e25 -R24/45/19/38 -JM16c -B
```
Comment (seisman) [inline test/seis/seis_13.sh:29]:
```suggestion
	sort -gr +9 tmp.txt | gmt meca -Sm0.4i+f0+l+s5e25 -R24/45/19/38 -JM16c -B
```

**Review by seisman — APPROVED** [review]
Looks good to me.

File: test/seis/seis_13.sh (hunk 1 above)
Comment (PaulWessel) [reply test/seis/seis_13.sh:29]: Sorry, forgot this one.  Just committed it to master...

**Pull request merged**